src/plugins/embeddable/public/lib/containers/container.ts

```diff
@@ -181,6 +181,7 @@ export abstract class Container<
         // If we hit this, the panel was removed before the embeddable finished loading.
         if (this.input.panels[id] === undefined) {
           subscription.unsubscribe();
+          // @ts-expect-error undefined in not assignable to TEmbeddable | ErrorEmbeddable
```

Comment:
we need either to change `untilEmbeddableLoaded` signature or fix the code @elastic/kibana-app-arch

Reply:
Maybe

```ts
reject(new Error('The panel was removed before the embeddable finished loading.'));
```

/cc @stacey-gammon